packages/models-library/src/models_library/projects_nodes.py

```diff
@@ -255,24 +272,114 @@ def _convert_empty_str_to_none(cls, v):
             return None
         return v
 
-    @classmethod
-    def _convert_old_enum_name(cls, v) -> RunningState:
-        if v == "FAILURE":
-            return RunningState.FAILED
-        return RunningState(v)
-
     @field_validator("state", mode="before")
     @classmethod
     def _convert_from_enum(cls, v):
         if isinstance(v, str):
+
+            def _convert_old_enum_name(v) -> RunningState:
+                if v == "FAILURE":
+                    return RunningState.FAILED
+                return RunningState(v)
```

Comment:
I would not define it each time inside the validator, consider moving it outs the class definition